runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -403,22 +400,18 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
          }
 
       J9ROMMethod *romMethod = (J9ROMMethod*)((uint8_t*)romClass + romMethodOffset);
-      J9Method *methodsOfClass = std::get<1>(classInfoTuple);
 
       // Build my entry
       if (!(optPlan = TR_OptimizationPlan::alloc(clientOptPlan.getOptLevel())))
          throw std::bad_alloc();
       optPlan->clone(&clientOptPlan);
 
-      TR::IlGeneratorMethodDetails *clientDetails = (TR::IlGeneratorMethodDetails*) &detailsStr[0];
-      *(uintptr_t*)clientDetails = 0; // smash remote vtable pointer to catch bugs early
-      TR::IlGeneratorMethodDetails details;
-      TR::IlGeneratorMethodDetails &remoteDetails = details.createRemoteMethodDetails(*clientDetails, detailsType, ramMethod, romClass, romMethod, clazz, methodsOfClass);
+      TR::IlGeneratorMethodDetails details(ramMethod);
```

Comment:
This factory routine should look very similar to `IlGeneratorMethodDetails::clone(TR::IlGeneratorMethodDetails &storage, const TR::IlGeneratorMethodDetails & other)` except that when determining the type you need to avoid calling virtual methods like `other.isOrdinaryMethod()` and instead have a parameter (bit field) that indicates which type of methodDetails we want to create.
On wrinkle is the use of `virtual getClass()` when creating newInstance thunks. Noah had that changed to a non-virtual  `classNeedingThunk()` method